docs/trusted-resources.md

```diff
@@ -0,0 +1,90 @@
+# Trusted Resources
+
+- [Overview](#overview)
+- [Instructions](#Instructions)
+ - [Sign Resources](#sign-resources)
+ - [Enable Trusted Resources](#enable-trusted-resources)
+
+## Overview
+
+Trusted Resources is a feature which can be used to sign Tekton Resources and verify them. Details of design can be found at [TEP--0091](https://github.com/tektoncd/community/blob/main/teps/0091-trusted-resources.md). This feature is under `alpha` version and support `v1beta1` version of `Task` and `Pipeline`.
+
+
+## Instructions
+
+### Sign Resources
+For `Sign` cli you may refer to [experimental repo](https://github.com/tektoncd/experimental/tree/main/pipeline/trusted-resources) to sign the resources. We're working to add `sign` and `verify` into [Tekton Cli](https://github.com/tektoncd/cli) as a subcommand.
+
+A signed task example:
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  annotations:
+    tekton.dev/signature: MEYCIQDM8WHQAn/yKJ6psTsa0BMjbI9IdguR+Zi6sPTVynxv6wIhAMy8JSETHP7A2Ncw7MyA7qp9eLsu/1cCKOjRL1mFXIKV
+  creationTimestamp: null
+  name: example-task
+  namespace: tekton-trusted-resources
+spec:
+  steps:
+  - image: ubuntu
+    name: echo
+```
+
+### Enable Trusted Resources
+
+#### Enable feature flag
+
+Update the config map:
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: feature-flags
+  namespace: tekton-pipelines
+  labels:
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/part-of: tekton-pipelines
+data:
+  enable-api-fields: "alpha"
+  resource-verification-mode: "enforce"
+```
+
+**Note:** `enable-api-fields` needs to be set as `alpha` and `resource-verification-mode` needs to be set as `enforce` or `warn` to enable resource verification.
+
+`resource-verification-mode` configurations:
+ * `enforce`: Failing verification will mark the taskruns/pipelineruns as failed.
+ * `warn`: Log warning but don't fail the taskruns/pipelineruns.
+ * `skip`: Directly skip the verification.
+
+Or patch the new values:
+```bash
+kubectl patch configmap feature-flags -n tekton-pipelines -p='{"data":{"enable-api-fields": "alpha", "resource-verification-mode":"enforce"}}
+```
+
+#### Config key at configmap
+Note that multiple keys reference should be separated by comma. If the resource can pass any key in the list, it will pass the verification.
+
+We currently hardcode SHA256 as hashfunc for loading public keys as verifiers.
+
+Public key files should be added into secret and mounted into controller volumes. To add keys into secret you may execute:
```

Comment:
Do we want to volume mount the secrets, or should we just reference them via the sigstore `k8s://namespace/name` syntax? https://docs.sigstore.dev/cosign/kubernetes

Reply:
Are you suggesting we use `cosign generate-key-pair k8s://default/testsecret` in docs? It's another way to save keys in secret, but we still need to mount this secret to controller container right?  Like what we do in Chains:
https://github.com/tektoncd/chains/blob/780521656ab0ea2a72d7a969a13040848ac6b969/config/100-deployment.yaml#L91-L92

Reply:
I was thinking more specifying these values in the config map - `publickeys: k8s://namespace/name, vault://key, gcpkms://key`. This way users don't need to redeploy Tekton each time they want to add a key.

I'm fine with punting on this as a TODO if you want, wanted to ask what the long term plan is since there's a few places where we assume local keys and my guess is that probably won't be the case forever.

Reply:
Oh I see, that seems better!
If we use `publickeys: k8s://namespace/name`, do we need to fetch secret to get the key files? 
In the long term I think we will still keep local keys as an option for users

Reply:
> @Yongxuanzhang is the updated plan for this reflected in the TEP?

Not yet, I will change this in the tep pr